opentelemetry-sdk/src/opentelemetry/sdk/trace/__init__.py

```diff
@@ -189,6 +189,45 @@ def attributes(self) -> types.Attributes:
         return self._event_formatter()
 
 
+def _is_valid_attribute_value(value: types.AttributeValue) -> bool:
+    """Checks if attribute value is valid.
+
+    An attribute value is valid if it is one of the valid types. If the value is a sequence, it is only valid
+    if all items in the sequence are of valid type, not a sequence, and are of the same type.
+    """
+
+    if isinstance(value, Sequence):
+        if len(value) == 0:
+            return True
+
+        first_element_type = type(value[0])
+
+        if first_element_type not in VALID_ATTR_VALUE_TYPES:
+            logger.warning(
+                "Invalid type %s in attribute value sequence. Expected one of %s or a sequence of those types",
+                first_element_type.__name__,
+                [valid_type.__name__ for valid_type in VALID_ATTR_VALUE_TYPES],
+            )
+            return False
+
+        for element in list(value)[1:]:
```

Comment:
Nit: it's a sequence, do you need to cast to slice it?

Reply:
Yes, if its a generator its not possible to slice it immediately

Reply:
Are generators sequences? It looks like they [need to implement __getitem__](https://docs.python.org/3/library/collections.abc.html#collections.abc.Sequence), which is what slice uses under the hood.